basis-library/integer/int-inf0.sml

```diff
@@ -1014,10 +1030,99 @@ structure IntInf =
                                  reserve (limbsFn (numLimbs lhs, numLimbs rhs), extra))
                 | SOME i => i
             end
+
+         (*
+          * quotient and remainder for small values (fitting in objptr)
+          * 
+          * Pull these functions out because each will be used twice,
+          * and we want to avoid redundant size checks if at all possible.
+          *)
+         fun smallQuot (num: bigInt, den: bigInt): bigInt =
+            let
+               val (numw, numi) = word_objptr_arg num
+               val (_, deni) = word_objptr_arg den
+            in
+               if numw = badObjptrWord 
+                  andalso deni = ~1
+               then negBadIntInf
+               else
+                  let
+                     val (_, ans) = word_tag_ans (I.quot (numi, deni))
+                  in 
+                     Prim.fromWord ans
+                  end
+            end
+
+         fun smallRem (num, den) =
+            let 
+               val (_, numi) = word_objptr_arg num
+               val (_, deni) = word_objptr_arg den
+               val (_, ans) = word_tag_ans (I.rem (numi, deni))
+            in 
+               Prim.fromWord ans
+            end
+
+         (*fun quotReserve (nlimbs, dlimbs) = reserve (S.- (nlimbs, dlimbs), 2)
+         fun remReserve (nlimbs, dlimbs) = reserve (dlimbs, 1)*)
       in
          val bigAdd = make (I.+!, Prim.+, S.max, 1)
          val bigSub = make (I.-!, Prim.-, S.max, 1)
          val bigMul = make (I.*!, Prim.*, S.+, 0)
+
+         fun bigQuot (num: bigInt, den: bigInt): bigInt =
+            if areSmall (num, den) then smallQuot (num, den)
+               else let
+                       val nlimbs = numLimbs num
+                       val dlimbs = numLimbs den
+                    in
+                       if S.< (nlimbs, dlimbs)
+                          then zero
+                          else if den = zero
+                                  then raise Div
+                                  else Prim.quot (num, den, reserve (S.- (nlimbs, dlimbs), 2))
```

Comment:
I think that we concluded that quot needs only `reserve (S.- (nlimbs, dlimbs), 1)`.

Reply:
Ok, I'll modify these as well as the quotRem reservations.